src/android/FCMPlugin.java

```diff
@@ -1,146 +1,256 @@
-						}catch(Exception e){
-							Log.d(TAG,"\tError retrieving token");
-						}
-					}
-				});
-			}
-			// NOTIFICATION CALLBACK REGISTER //
-			else if (action.equals("registerNotification")) {
-				notificationCallBackReady = true;
-				cordova.getActivity().runOnUiThread(new Runnable() {
-					public void run() {
-						if(lastPush != null) FCMPlugin.sendPushPayload( lastPush );
-						lastPush = null;
-						callbackContext.success();
-					}
-				});
-			}
-			// UN/SUBSCRIBE TOPICS //
-			else if (action.equals("subscribeToTopic")) {
-				cordova.getThreadPool().execute(new Runnable() {
-					public void run() {
-						try{
-							FirebaseMessaging.getInstance().subscribeToTopic( args.getString(0) );
-							callbackContext.success();
-						}catch(Exception e){
-							callbackContext.error(e.getMessage());
-						}
-					}
-				});
-			}
-			else if (action.equals("unsubscribeFromTopic")) {
-				cordova.getThreadPool().execute(new Runnable() {
-					public void run() {
-						try{
-							FirebaseMessaging.getInstance().unsubscribeFromTopic( args.getString(0) );
-							callbackContext.success();
-						}catch(Exception e){
-							callbackContext.error(e.getMessage());
-						}
-					}
-				});
-			}
-			else{
-				callbackContext.error("Method not found");
-				return false;
-			}
-		}catch(Exception e){
-			Log.d(TAG, "ERROR: onPluginAction: " + e.getMessage());
-			callbackContext.error(e.getMessage());
-			return false;
-		}
-		
-		//cordova.getThreadPool().execute(new Runnable() {
-		//	public void run() {
-		//	  //
-		//	}
-		//});
-		
-		//cordova.getActivity().runOnUiThread(new Runnable() {
-        //    public void run() {
-        //      //
-        //    }
-        //});
-		return true;
-	}
-	
-	public static void sendPushPayload(Map<String, Object> payload) {
-		Log.d(TAG, "==> FCMPlugin sendPushPayload");
-		Log.d(TAG, "\tnotificationCallBackReady: " + notificationCallBackReady);
-		Log.d(TAG, "\tgWebView: " + gWebView);
-	    try {
-		    JSONObject jo = new JSONObject();
-			for (String key : payload.keySet()) {
-			    jo.put(key, payload.get(key));
-				Log.d(TAG, "\tpayload: " + key + " => " + payload.get(key));
+  public FCMPlugin() {}
+
+  public void initialize(CordovaInterface cordova, CordovaWebView webView) {
+    super.initialize(cordova, webView);
+    gWebView = webView;
+    Log.d(TAG, "==> FCMPlugin initialize");
+    FirebaseMessaging.getInstance().subscribeToTopic("android");
+    FirebaseMessaging.getInstance().subscribeToTopic("all");
+  }
+
+  public boolean execute(final String action, final JSONArray args, final CallbackContext callbackContext) throws JSONException {
+
+    Log.d(TAG,"==> FCMPlugin execute: "+ action);
+
+    try{
+      // READY //
+      if (action.equals("ready")) {
+        callbackContext.success();
+      }
+      // GET VERIFICATION ID //
+      else if (action.equals("getVerificationID")) {
+        cordova.getActivity().runOnUiThread(new Runnable() {
```

Comment:
Why the change from the other persons source, where it was `cordova.getThreadPool().execute(new Runnable() { ...`